### PR TITLE
stf_lib CTest regression

### PR DESCRIPTION
**Pull request by StanSofomoDev**
This pull request introduces a regression target for unit testing STF_LIB using CTest. The regression tests can be executed with the following commands:

```sh
mkdir -p release/
cd release/
cmake .. -DCMAKE_BUILD_TYPE=Release
make -j32 regress
cd ..
```
Currently, these tests serve as a stub and do not cover the STF_LIB extensively (only one basic STFWriter test was added).

### CODE REVIEW

File: tests/stf_writer_test/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+add_compile_options(
+    -g
+    -Wall
+    -Wno-parentheses
+    -MMD
+    -D_FILE_OFFSET_BITS=64
+    -D_LARGEFILE_SOURCE
+    -D_GNU_SOURCE
+    -D__STDC_FORMAT_MACROS
+)
```
Comment (klingaard) [inline tests/stf_writer_test/CMakeLists.txt:23]: Some others to consider:
```
  -Werror -Wdeprecated -Wextra -Winline -Winit-self -Wno-unused-function
  -Wuninitialized -Wno-sequence-point -Wno-inline -Wno-unknown-pragmas
  -Woverloaded-virtual -Wno-unused-parameter -Wno-missing-field-initializers
```

File: tests/stf_writer_test/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+add_compile_options($<$<CONFIG:Release>:-Ofast>)
```
Comment (klingaard) [inline tests/stf_writer_test/CMakeLists.txt:25]: Wonder if LTO would be good to test for Release?  @bdutro

**Review by klingaard — COMMENTED** [review]
This is great! And appreciated.  One thing this doesn't really test is making sure the file written contains the expected information written.  Suggest adding an `stf_reader` after the close and making sure the file read matches what was written.

**Comment by StanSofomoDev** [comment]
> This is great! And appreciated. One thing this doesn't really test is making sure the file written contains the expected information written. Suggest adding an `stf_reader` after the close and making sure the file read matches what was written.

Written file contents are tested by diff comparison to golden file that is included in the test directory.

File: tests/stf_writer_test/CMakeLists.txt (hunk 1 above)
Comment (StanSofomoDev) [reply tests/stf_writer_test/CMakeLists.txt:36]: Done.

File: tests/stf_writer_test/CMakeLists.txt (hunk 2 above)
Comment (StanSofomoDev) [reply tests/stf_writer_test/CMakeLists.txt:38]: Done.

**Comment by StanSofomoDev** [comment]
@bdutro, @klingaard, main purpose for this PR is to add CTest environment to the `stf_lib`, proper tests that would cover components of the `stf_lib` can be added in the future.

**Comment by bdutro** [comment]
This all looks good. Could you go ahead and add a `make regress` call to https://github.com/sparcians/stf_lib/blob/master/.github/workflows/ubuntu-build.yml so it runs for PRs?

**Review by bdutro — APPROVED** [review]
LGTM

**Pull request merged**